rollout/trafficrouting_test.go

```diff
@@ -292,18 +305,13 @@ func TestRolloutWithExperimentStep(t *testing.T) {
 			// make sure SetWeight was called with correct value
 			assert.Equal(t, int32(10), desiredWeight)
 			assert.Equal(t, int32(5), weightDestinations[0].Weight)
+			assert.Len(t, weightDestinations, 1)
 			assert.Equal(t, ex.Status.TemplateStatuses[0].ServiceName, weightDestinations[0].ServiceName)
 			assert.Equal(t, ex.Status.TemplateStatuses[0].PodTemplateHash, weightDestinations[0].PodTemplateHash)
 			return nil
 		})
 		f.fakeTrafficRouting.On("SetHeaderRoute", mock.Anything, mock.Anything).Return(nil)
-		f.fakeTrafficRouting.On("VerifyWeight", mock.Anything).Return(func(desiredWeight int32, weightDestinations ...v1alpha1.WeightDestination) error {
```

Comment:
Why are these test cases being removed?

Reply:
@zachaller   because these assertions are already present here : https://github.com/argoproj/argo-rollouts/pull/2734/files#:~:text=with%20correct%20value-,assert.Equal(t%2C%20int32(10)%2C%20desiredWeight,%5B0%5D.PodTemplateHash%2C%20weightDestinations%5B0%5D.PodTemplateHash),-return%20nil

and also, these assertions were not being executed till now.. an exception was being thrown at the beginning of this UT but somehow this UT was not failing .. hence, I updated this existing UT itself so that these assertions are properly executed.